client/src/components/map/Map.tsx

```diff
@@ -1,188 +1,492 @@
+
+  useEffect(() => {
+    if (!mapRef.current) return;
+
+    const map = mapRef.current;
+    map.setZoom(props.zoom || defaultZoom);
+  }, [props.zoom]);
+
+  return (
+    <div
+      className={props.className}
+      ref={mapContainerRef}
+      id="map"
+      style={{
+        ...props.style,
+        ...{ pointerEvents: "none", touchAction: "none" },
+      }}
+    ></div>
+  );
+};
+
+export const DashboardMap: React.FC<SatMapProps> = (props) => {
+  const mapContainerRef = useRef<any>(null);
+  const mapRef = useRef<mapboxgl.Map | null>(null);
+
+  useEffect(() => {
+    if (mapRef.current) return;
+
+    const modes = MapboxDraw.modes;
+    modes.static = StaticMode;
+
+    mapRef.current = new mapboxgl.Map({
+      container: mapContainerRef.current,
+      style: "mapbox://styles/mapbox/satellite-streets-v12",
+      center: center,
+      zoom: props.zoom || defaultZoom,
+      pitch: 40,
+      interactive: true,
+    });
+
+    mapRef.current.addControl(new mapboxgl.ScaleControl(), "bottom-right");
+    mapRef.current.addControl(new mapboxgl.NavigationControl(), "bottom-right");
+    mapRef.current.addControl(new mapboxgl.FullscreenControl(), "bottom-right");
+    mapRef.current.addControl(DashboardMapDraw, "bottom-right");
+
+    mapRef.current.on("load", function () {
+      DashboardMapDraw.changeMode("static");
+    });
+
+    if (props.drawing) {
+      DashboardMapDraw.set(props.drawing);
+    }
+  }, [mapContainerRef.current]);
+
+  useEffect(() => {
+    if (props.drawing) {
+      mapRef.current?.remove();
+      mapRef.current = null;
+
+      const modes = MapboxDraw.modes;
+      modes.static = StaticMode;
+
+      mapRef.current = new mapboxgl.Map({
+        container: mapContainerRef.current,
+        style: "mapbox://styles/mapbox/satellite-streets-v12",
+        center: center,
+        zoom: props.zoom || defaultZoom,
+        pitch: 40,
+        interactive: true,
+      });
+
+      mapRef.current.addControl(new mapboxgl.ScaleControl(), "bottom-right");
+      mapRef.current.addControl(
+        new mapboxgl.NavigationControl(),
+        "bottom-right"
+      );
+      mapRef.current.addControl(
+        new mapboxgl.FullscreenControl(),
+        "bottom-right"
+      );
+      mapRef.current.addControl(DashboardMapDraw, "bottom-right");
+
+      mapRef.current.on("load", function () {
+        DashboardMapDraw.changeMode("static");
+      });
+
+      if (props.drawing) {
+        DashboardMapDraw.set(props.drawing);
+      }
+
+      // Load custom icon for points
+      mapRef.current.on("load", () => {
+        mapRef.current?.loadImage(
+          `${window.location.origin}/pin.png`,
+          (error, image: any) => {
+            if (error) throw error;
+            if (!mapRef.current?.hasImage("pin-icon")) {
+              mapRef.current?.addImage("pin-icon", image);
+            }
+
+            // Add points to map from props.drawing
+            mapRef.current?.addSource("documents", {
```

Comment:
try to filter for drawing with type:point and not polygon here to avoid plotting points in areas